### PR TITLE
add responsive parameter and set to true

### PR DESCRIPTION
**Pull request by entonbiba**
Responsive waveform on resize and device orientation change set to true by default

parameter `responsive` default `true`, can be set to `false` to disable responsive waveform.

**todo**
- create onResize method
- add event handler 
`this._onResize = this.onResize.bind(this)`

### CODE REVIEW

**Comment by thijstriemstra** [comment]
Can you place this in a separate method, not add it to `init`?

**Comment by katspaugh** [comment]
I would define `onResize` as a regular method, and add `this._onResize = this.onResize.bind(this)` for the event handler.

**Comment by X-Raym** [comment]
@entonbiba does this means that code snippets to resize the waveform on window resize event like you shew me here https://github.com/katspaugh/wavesurfer.js/issues/1067#issuecomment-297217766 will not required anymore ?

**Comment by entonbiba** [comment]
@X-Raym yes

**Comment by entonbiba** [comment]
@katspaugh I'll update this code on the weekend

File: src/wavesurfer.js [hunk 1]
```diff
@@ -84,6 +85,23 @@ var WaveSurfer = {
...
+        // Responsive waveform on resize and device orientation change
+        // the waveform will be responsive by default
+        // if the responsive parameter is not set or set to true
+        // setting the responsive paramater to false will disable the responsive waveform
+        var prevWidth = 0;
+        this.onResize = WaveSurfer.util.debounce(function () {
+            if (prevWidth != this.wrapper.clientWidth) {
+                prevWidth = this.wrapper.clientWidth;
+                this.render();
+                this.progress(WaveSurfer.backend.getPlayedPercents());
```
Comment (tf) [inline src/wavesurfer.js]: I could not find the definition of neither `render` nor `progress`. I think calling `drawBuffer` should be enough once the redrawing bug discussed in #1127 is resolved.

**Comment by X-Raym** [comment]
@entonbiba Any news from this feature ?
Also, does this feature also adjust the cursor potion in the waveform to match the new waveform ?

Thanks !

**Comment by mspae** [comment]
@X-Raym This PR was merged added to v2 some time ago. It currently does not support correct handling of position, this should have been fixed in #1210 but since that PR will probably not be merged this needs to be fixed in a seperate PR.

**Comment by X-Raym** [comment]
@mspae
ok ! this feature was intersting, I hope it will be bump soon.

**Comment by mspae** [comment]
@entonbiba Can we merge into `es5`  or close? Since #1219 this functionality is present in v2.

**Comment by mspae** [comment]
I'm closing this. Please comment/reopen if anybody disagrees.

**Pull request closed without merging**